hail/python/hail/experimental/db.py

```diff
@@ -21,27 +21,41 @@ class DatasetVersion:
     :class:`DatasetVersion` has two constructors: :func:`.from_json` and :func:`.get_region`.
     """
     @staticmethod
-    def from_json(doc: dict) -> 'DatasetVersion':
+    def from_json(doc: dict, cloud: str, custom_config: bool) -> 'DatasetVersion':
         """Create :class:`.DatasetVersion` object from dictionary.
 
         Parameters
         ----------
         doc : :obj:`dict`
-            Dictionary containing url and version keys.
+            Dictionary containing url and version keys. Value for url is a either
+            a :obj:`str` with a url, or a :obj:`dict` containing key: value pairs,
+            like region: url.
+        cloud : :obj:`str`
+            Cloud platform to access dataset, either 'gcp' or 'aws'. Not used
+            here if using a custom configuration.
+        custom_config : :obj:`bool`
+            Boolean indicating whether or not dataset is from a :class:`.DB` object
+            using a custom configuration or url.
 
         Returns
         -------
         :class:`.DatasetVersion`
         """
         assert 'url' in doc, doc
         assert 'version' in doc, doc
-        return DatasetVersion(doc['url'],
-                              doc['version'])
+        assert 'reference_genome' in doc, doc
+        if not custom_config:
+            assert cloud in doc['url'], doc['url']
+            url = doc['url'][cloud]
+        else:
+            url = doc['url']
```

Comment:
I'm inclined to require custom configs to have the same format as the publicly supported one. AFAIK, nobody is using this functionality. Moreover, this is in the `experimental` module, so we have no obligation to maintain compatibility.

Do you have an argument in favor of different formats for the custom vs non-custom? I suppose a custom config is probably used by one lab which will exist in one cloud. I can see its a bit more annoying, but I prefer the consistency.

Reply:
Not really any argument for different formats, and the consistency would make things nicer. With that in mind, I think I'll also modify `Dataset.from_name_and_json()`, as it currently doesn't check for region with a custom config. That way the expected formatting for a custom config will be consistent with whats in `datasets.json`.